src/bp-forums/common/widgets.php

```diff
@@ -591,7 +591,7 @@ public function widget( $args, $instance ) {
 			array(
 				'post_type'           => bbp_get_forum_post_type(),
 				// 'post_parent'         => $settings['parent_forum'],
-				'post_parent'         => 0,
+				'post_parent'         => intval( $settings['parent_forum'] ),
```

Comment:
@kent-carlo I think you need to handle case for `$settings['parent_forum'] = 'any'` here.